src/payment/strategies/googlepay/googlepay-payment-processor.spec.ts

```diff
@@ -238,6 +238,56 @@ describe('GooglePayPaymentProcessor', () => {
             expect(googlePayInitializer.parseResponse).toHaveBeenCalled();
         });
 
+        it('handles success response with 1 word in fullname', async () => {
+            jest.spyOn(googlePayInitializer, 'parseResponse').mockReturnValue(tokenizePayload);
+            jest.spyOn(requestSender, 'post').mockReturnValue(Promise.resolve());
+            jest.spyOn(store.getState().billingAddress, 'getBillingAddress').mockReturnValue(getGooglePayAddressMock());
+            const googlePaymentDataMock = getGooglePaymentDataMock();
+            googlePaymentDataMock.paymentMethodData.info.billingAddress.name = 'mock';
+
+            await processor.initialize('googlepay');
+            await processor.handleSuccess(googlePaymentDataMock);
+
+            expect(googlePayInitializer.parseResponse).toHaveBeenCalled();
+        });
+
+        it('handles success response with 2 word in fullname', async () => {
+            jest.spyOn(googlePayInitializer, 'parseResponse').mockReturnValue(tokenizePayload);
+            jest.spyOn(requestSender, 'post').mockReturnValue(Promise.resolve());
+            jest.spyOn(store.getState().billingAddress, 'getBillingAddress').mockReturnValue(getGooglePayAddressMock());
+            const googlePaymentDataMock = getGooglePaymentDataMock();
+
+            await processor.initialize('googlepay');
+            await processor.handleSuccess(googlePaymentDataMock);
+
+            expect(googlePayInitializer.parseResponse).toHaveBeenCalled();
+        });
+
+        it('handles success response with 3 word in fullname', async () => {
+            jest.spyOn(googlePayInitializer, 'parseResponse').mockReturnValue(tokenizePayload);
+            jest.spyOn(requestSender, 'post').mockReturnValue(Promise.resolve());
+            jest.spyOn(store.getState().billingAddress, 'getBillingAddress').mockReturnValue(getGooglePayAddressMock());
+            const googlePaymentDataMock = getGooglePaymentDataMock();
+            googlePaymentDataMock.paymentMethodData.info.billingAddress.name = 'mock mock mock';
```

Comment:
These tests don't prove that `_getFirstAndLastName` works as intended. If we change this value from `mock mock mock` to something else, the test still passes. 

Instead, if we make `getFirstAndLastName` an imported utility dependancy of this file, we can test it more directly.

E.g. something like:

```
import { getFirstAndLastName } from './get-first-and-last-name';

describe('With a one part name', () => {
    it('assigns the whole string as the first name', () => {
        expect(getFirstAndLastName('Cher')).toEqual(['Cher', '']);
    });
});

describe('With a two part name', () => {
    it('splits the string into a first and a second name', () => {
        expect(getFirstAndLastName('Lee Dunkley')).toEqual(['Lee', 'Dunkley']);
    });
});

describe('With a three or more part name', () => {
    it('assigns the final part as the last name and all others as the first name', () => {
        expect(getFirstAndLastName('Neil deGrasse Tyson')).toEqual(['Neil deGrasse', 'Tyson']);
        expect(getFirstAndLastName('Ignacio Arsenio Travieso Scull')).toEqual(['Ignacio Arsenio Travieso', 'Scull']);
    });
});
```

Then this file (the class' tests) can be ignorant of all the permutations and just test one representative variant.